src/compas/robots/model/joint.py

```diff
@@ -427,14 +416,24 @@ def calculate_planar_transformation(self, position):
     def calculate_transformation(self, position):
         """Returns the transformation of the joint.
 
-        This function is overwitten in the init based on the joint type.
+        This function calls different calculate_*_transformation depends on self.type
 
         Parameters
         ----------
         position : :obj:`float`
             Position in radians or meters depending on the joint type.
         """
-        pass
+
+        switcher = {
+            Joint.REVOLUTE: self.calculate_revolute_transformation,
+            Joint.CONTINUOUS: self.calculate_continuous_transformation,
+            Joint.PRISMATIC: self.calculate_prismatic_transformation,
+            Joint.FIXED: self.calculate_fixed_transformation,
+            Joint.FLOATING: self.calculate_floating_transformation,
+            Joint.PLANAR: self.calculate_planar_transformation
+        }
+        # set the transformation function according to the type
+        return switcher.get(self.type)(position)
```

Comment:
To avoid switching constantly, we cache on first call.

```suggestion
        # Set the transformation function according to the type
        if not self._calculate_transformation:
            switcher = {
                Joint.REVOLUTE: self.calculate_revolute_transformation,
                Joint.CONTINUOUS: self.calculate_continuous_transformation,
                Joint.PRISMATIC: self.calculate_prismatic_transformation,
                Joint.FIXED: self.calculate_fixed_transformation,
                Joint.FLOATING: self.calculate_floating_transformation,
                Joint.PLANAR: self.calculate_planar_transformation
            }
            self._calculate_transformation = switcher.get(self.type)

        return self._calculate_transformation(position)
```